model/jpa/src/main/java/org/keycloak/connections/jpa/updater/liquibase/conn/CustomChangeLogHistoryService.java

```diff
@@ -0,0 +1,120 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.connections.jpa.updater.liquibase.conn;
+
+import org.keycloak.common.util.reflections.Reflections;
+import java.lang.reflect.Field;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import liquibase.ContextExpression;
+import liquibase.Labels;
+import liquibase.change.CheckSum;
+import liquibase.changelog.ChangeSet;
+import liquibase.changelog.RanChangeSet;
+import liquibase.changelog.StandardChangeLogHistoryService;
+import liquibase.database.Database;
+import liquibase.database.core.MySQLDatabase;
+import liquibase.exception.DatabaseException;
+import liquibase.logging.LogFactory;
+
+/**
+ *
+ * @author hmlnarik
+ */
+public class CustomChangeLogHistoryService extends StandardChangeLogHistoryService {
+
+    private List<RanChangeSet> ranChangeSetList;
+
+    @Override
+    public List<RanChangeSet> getRanChangeSets() throws DatabaseException {
+        Database database = getDatabase();
+        if (! (database instanceof MySQLDatabase) || database.getDatabaseMajorVersion() < 8) {
+            return super.getRanChangeSets();
+        }
+        if (this.ranChangeSetList == null) {
+            String databaseChangeLogTableName = getDatabase().escapeTableName(getLiquibaseCatalogName(), getLiquibaseSchemaName(), getDatabaseChangeLogTableName());
+            List<RanChangeSet> ranChangeSetList = new ArrayList<>();
+            if (hasDatabaseChangeLogTable()) {
+                LogFactory.getLogger().info("Reading from " + databaseChangeLogTableName);
+                List<Map<String, ?>> results = queryDatabaseChangeLogTable(database);
+                for (Map rs : results) {
+                    String fileName = rs.get("FILENAME").toString();
+                    String author = rs.get("AUTHOR").toString();
+                    String id = rs.get("ID").toString();
+                    String md5sum = rs.get("MD5SUM") == null || getDatabaseChecksumsCompatible() ? null : rs.get("MD5SUM").toString();
+                    String description = rs.get("DESCRIPTION") == null ? null : rs.get("DESCRIPTION").toString();
+                    String comments = rs.get("COMMENTS") == null ? null : rs.get("COMMENTS").toString();
+                    Object tmpDateExecuted = rs.get("DATEEXECUTED");
+                    Date dateExecuted = null;
+                    if (tmpDateExecuted instanceof Date) {
+                        dateExecuted = (Date) tmpDateExecuted;
+                    } else if (tmpDateExecuted instanceof LocalDateTime) {
+                        dateExecuted = Date.from(((LocalDateTime) tmpDateExecuted).atZone(ZoneId.systemDefault()).toInstant());
```

Comment:
This is the only difference as compared to `StandardChangeLogHistoryService.getRanChangeSets`, and is based on the original fix in liquibase: https://github.com/liquibase/liquibase/commit/90c523562fc6f502d439ba1670bb0793a6467a6c